src/api/libcellml/component.h

```diff
@@ -37,12 +37,19 @@ class LIBCELLML_EXPORT Component: public ComponentEntity, public ImportedEntity
                                   public std::enable_shared_from_this<Component>
 #endif
 {
-public:
+private:
     Component(); /**< Constructor */
+
+public:
     ~Component() override; /**< Destructor */
-    Component(const Component &rhs); /**< Copy constructor */
-    Component(Component &&rhs) noexcept; /**< Move constructor */
-    Component &operator=(Component rhs); /**< Assignment operator */
+    Component(const Component &rhs) = delete; /**< Copy constructor */
+    Component(Component &&rhs) noexcept = delete; /**< Move constructor */
+    Component &operator=(Component rhs) = delete; /**< Assignment operator */
+
+    template<typename... Args>
```

Comment:
I haven't looked at the rationale for now using templates, but... I thought we wanted to avoid having to do that "at all costs"?...